### ISSUE TITLE
NPE occurred when use database

### ISSUE THREAD

**Issue by zjcnb** [apache/shardingsphere#17944]
### Which version of ShardingSphere did you use?

`master`

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

`ShardingSphere-Proxy`

```
[ERROR] 2022-05-25 18:27:45.967 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
	at org.apache.shardingsphere.proxy.backend.response.header.query.QueryHeaderBuilderFactory.getInstance(QueryHeaderBuilderFactory.java:44)
	at org.apache.shardingsphere.proxy.backend.response.header.query.QueryHeaderBuilderEngine.build(QueryHeaderBuilderEngine.java:55)
	at org.apache.shardingsphere.proxy.backend.text.admin.DatabaseAdminQueryBackendHandler.createResponseHeader(DatabaseAdminQueryBackendHandler.java:72)
	at org.apache.shardingsphere.proxy.backend.text.admin.DatabaseAdminQueryBackendHandler.execute(DatabaseAdminQueryBackendHandler.java:61)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:97)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:106)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[ERROR] 2022-05-25 18:27:45.986 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
	at org.apache.shardingsphere.proxy.backend.response.header.query.QueryHeaderBuilderFactory.getInstance(QueryHeaderBuilderFactory.java:44)
	at org.apache.shardingsphere.proxy.backend.response.header.query.QueryHeaderBuilderEngine.build(QueryHeaderBuilderEngine.java:55)
	at org.apache.shardingsphere.proxy.backend.text.admin.DatabaseAdminQueryBackendHandler.createResponseHeader(DatabaseAdminQueryBackendHandler.java:72)
	at org.apache.shardingsphere.proxy.backend.text.admin.DatabaseAdminQueryBackendHandler.execute(DatabaseAdminQueryBackendHandler.java:61)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:97)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:106)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### server.yaml

```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: governance_ds
      server-lists: localhost:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: false

rules:
  - !AUTHORITY
    users:
      - root@%:root
      - sharding@:sharding
    provider:
      type: ALL_PERMITTED
```

<img width="603" alt="image" src="https://user-images.githubusercontent.com/33742097/170242412-afc0a683-f4ba-4fb8-8b5c-29f59f71bc46.png">

**Comment by zjcnb** [comment]
@TeslaCN Hi, can you look this?

**Comment by strongduanmu** [comment]
@zjcnb I meet the same exception when execute `show tables`. I will try to fix this bug.

**Comment by zjcnb** [comment]
> @zjcnb I meet the same exception when execute `show tables`. I will try to fix this bug.

OK,  thx

**Comment by strongduanmu** [comment]
Fixed by #17944.
